### DISCUSSION TITLE
Coinbase Wallet not detected on IOS (Safari)

### DISCUSSION THREAD

**Discussion by migbash** [WalletConnect/walletconnect-monorepo#5380] [Developer Support]
Coinbase Wallet is not being detected on IOS (Safari), despite the applicatication being available on the mobile.

<img
src=https://github.com/user-attachments/assets/0225950d-bf54-4a44-b876-ffc04ecaa705
width=250
/>

How to get it working ? It's the only wallet that does not seem to work corretly on `WalletConnect`.

Works correctly on Desktop with the `Coinbase Wallet Extension`.

**Comment by arein** [comment]
Safari has no API to detect which applications you have installed on your phone.

On desktop extensions follow the EIP-6963 standard reown did last year to inject and advertise their support.

On native apps we do show installed wallets.

**Reply by migbash** [reply]
@arein Huh ? But it works for `MetaMask` and `TrustWallet` on `IOS` via Safari... I do not understand why `Coinbase` is not picked up also.

<img
src=https://github.com/user-attachments/assets/b5d97107-1a73-444d-a00c-0097b9fd3c99
width=250
/>

<img
src=https://github.com/user-attachments/assets/4bcbfdb2-e28f-4e41-a5d5-9a405ed295a1
width=250
/>

I was using the packages: `"@walletconnect/web3-provider": "1.8.0", "@web3modal/ethers5": "3.5.7"` before, and just recently upgraded to the latest `reown` packages respectively for `appkit` and its still not working for `Coinbase Wallet` detection on `Mobile` for `IOS`.

**Reply by arein** [reply]
I think I may have misunderstood. Can you please with screenshots show the actual vs expected behavior.

There is no indication that Trust is installed. So maybe that language was misleading me.

**Reply by migbash** [reply]
@arein No worries.

All respective `mobile` applications: (1) MetaMask, (2) TrustWallet and (3) Coinbase Wallet are installed on my device.

When I use either (1) MetaMask or (2) TrustWallet I get promted to be redirected to respetive applications for Authentication stuff to be completed by the user (what currently actually happens and is correct):

<img
src=https://github.com/user-attachments/assets/0d1cb189-cde5-418a-aacd-7936fa42ff4a
width=250
/>

<img
src=https://github.com/user-attachments/assets/0382cc40-1491-47d8-93d6-7ef668e26186
width=250
/>

However, this step is not working with Coinbase Wallet on **MOBILE** wallet option on Walletconnect, saying the following:

<img
src=https://github.com/user-attachments/assets/0225950d-bf54-4a44-b876-ffc04ecaa705
width=250
/>

despite the Coinbase Wallet being installed and active on my phone. Again, I want to note that the Coinbase Wallet Chrome extension works fine with walletconnect, only issue is for its `mobile` logic.

What I am expecting to happen is the same as with the `Metamask` example screenshots I have shown, I want to be redirected to the `Coinbase Wallet` mobile application, but it is not detecting the application on my phone, like its doing with `Metamask` and `Trustwallet` correctly.

**Reply by migbash** [reply]
@arein ? Any update on this ?

**Reply by arein** [reply]
https://github.com/user-attachments/assets/3b966309-19b5-47a7-98d1-c1408644d99a

What is your configuration - for me it works as expected - do you set Coinbase as [`featuredWallet`](https://docs.reown.com/appkit/shared/options#featuredwalletids) in the AppKit config?

I am using https://appkit-lab.reown.com/library/wagmi/ on iOS Safari in the above video
